dotnet/src/Connectors/Connectors.Memory.AzureCognitiveSearch/AzureCognitiveSearchMemoryStore.cs

```diff
@@ -428,5 +429,16 @@ private static T RunMemoryStoreOperation<T>(Func<T> operation)
         }
     }
 
+    private static double ScoreToCosineSimilarity(double score)
+    {
+        score = Math.Max(score, 1.0 / 3);
+        return 1 - (1 - score) / score;
+    }
```

Comment:
Not sure where to find prior conversation if this already was anwered...
Why use `return 1 - (1 - score) / score;` instead of `return 2 - 1 / score;`  ?

Reply:
I see response in email that:
`1 - (1 - distance) / distance` is `1 - ((1 - distance) / distance) ` is `(2 * distance -1) / distance`  Agreed.  And this is `2 - (1 / distance)`  which is  `2 - 1 / distance`.

Why do we need to evaluate distance twice in the expression?  It can be evaluated once.